packages/controller/src/lib/notificationHandler.ts

```diff
@@ -4,24 +4,117 @@
  *  2021 foxriver76 <moritz.heusinger@gmail.com>
  */
 
-const fs = require('fs-extra');
-const { tools } = require('@iobroker/js-controller-common');
-const path = require('path');
+import fs from 'fs-extra';
+import { tools } from '@iobroker/js-controller-common';
+import path from 'path';
+import type { Client as StatesInRedisClient } from '@iobroker/db-states-redis';
+import type { Client as ObjectsInRedisClient } from '@iobroker/db-objects-redis';
+import type Winston from 'winston';
+
+export interface NotificationHandlerSettings {
+    host: string;
+    states: StatesInRedisClient;
+    objects: ObjectsInRedisClient;
+    log: Winston.Logger;
+    logPrefix: string;
+}
 
-class NotificationHandler {
-    /**
-     * Create a new instance of NotificationHandler
-     *
-     * @param {object} settings - settings like states/objects db
-     */
-    constructor(settings) {
+interface MultilingualObject {
+    en: string;
+    [otherLanguages: string]: string;
+}
+
+export interface NotificationsConfigEntry {
+    /** e.g. system */
+    scope: string;
+    /** multilingual name */
+    name: MultilingualObject;
+    /** multilingual description */
+    description: MultilingualObject;
+    categories: CategoryConfigEntry[];
+}
+
+export type Severity = 'info' | 'notify' | 'alert';
+
+export interface CategoryConfigEntry {
+    category: string;
+    name: MultilingualObject;
+    /** `info` will only be shown by admin, while `notify` might also be used by messaging adapters, `alert` ensures both */
+    severity: Severity;
+    description: MultilingualObject;
+    regex: string[];
+    limit: number;
+}
+
+interface NotificationMessageObject {
+    message: string;
+    ts: number;
+}
+
+interface NotificationsObject {
+    [scope: string]: {
+        [category: string]: {
+            [instance: string]: NotificationMessageObject[];
+        };
+    };
+}
+
+export interface FilteredNotificationInformation {
+    [scope: string]: {
+        description: MultilingualObject;
+        name: MultilingualObject;
+        categories: {
+            [category: string]: {
+                description: MultilingualObject;
+                name: MultilingualObject;
+                severity: Severity;
+                instances: {
+                    [instance: string]: {
+                        messages: NotificationMessageObject[];
+                    };
+                };
+            };
+        };
+    };
+}
+
+interface NotificationSetupCategory {
+    regex: RegExp[];
+    limit: number;
+    name: MultilingualObject;
+    severity: Severity;
+    description: MultilingualObject;
+}
+
+interface NotificationSetupObject {
+    [scope: string]: {
+        name: MultilingualObject;
+        description: MultilingualObject;
+        categories: Record<string, NotificationSetupCategory>;
```

Comment:
This is a bit inconsistent with the other nested objects which use an index signature.